### PR TITLE
feat(breaking): use 'podspecPath' instead of 'podspec', make it smarter

### PR DESCRIPTION
**Pull request by thymikee**
Summary:
---------

This diff removes the `project.ios.podspec` in favor of more obvious `project.ios.podspecPath` which makes it possible for users to provide a custom path to the `.podspec` file. I chose this, as it aligns better with what we have already in iOS config (namely `projectPath` and `pbxprojPath`).

I also made searching for the podspec a bit smarter – by not only looking in the library root, but also in its `sourceDir`. With this, I could probably make this change non-breaking, because it covers a lot more libs with podspec resolution. However, the `podspec` as it was (just the name of a podspec) kinda doesn't make much sense and is pretty limited. I think path makes it more obvious.

Also the only user of `podspec` option I know of is @Salakar – let me know how this works for you.

Fixes https://github.com/react-native-community/cli/issues/412

cc @orta @alloy 

Test Plan:
----------

Adjusted existing tests to cover this functionality.

### CODE REVIEW

File: packages/platform-ios/src/config/index.js [hunk 1]
```diff
@@ -47,13 +47,19 @@ export function projectConfig(
...
+    sourceDir,
     folder,
     pbxprojPath: path.join(projectPath, 'project.pbxproj'),
     podfile: findPodfilePath(projectPath),
-    podspec: userConfig.podspec || findPodspecName(folder),
+    podspecPath:
+      userConfig.podspecPath ||
+      // podspecs are usually placed in the root dir of the library or in the
+      // iOS project path
+      findPodspec(folder) ||
```
Comment (grabbou) [inline packages/platform-ios/src/config/index.js]: I wonder whether the order is correct. `sourceDir` is within `folder`, so theoretically speaking, `findPodspec(folder)` will always return the same results as `findPodspec(sourceDir)` and never return `null`.

Comment (thymikee) [reply packages/platform-ios/src/config/index.js]: I thought the same but that's not how it works, the glob for podspec is `glob('*.podspec', {cwd})` which only traverses the cwd and goes no further. I concluded this as desired behavior for performance reasons (plus it happens that there are node_modules inside that cwd, and unrelated podspecs as well)

Comment (grabbou) [reply packages/platform-ios/src/config/index.js]: Okay, so it only looks at the root level and doesn't go any deeper. Smart.

**Comment by alloy** [comment]
Yup, agreed, much better to use full path 👍

**Pull request merged**